### DISCUSSION TITLE
Set absolute path for assetsfolder

### DISCUSSION THREAD

**Discussion by JessinhoW** [stephanrauh/ngx-extended-pdf-viewer#897] [Q&A]
Hi,
We use our own CDN and uses the pdfviewer in different applications. When I set the assets folder to the http url of where we host the pdfjs folder it is not working because it keeps putting the current application url before the real assets folder url. Is there a way to set the assets folder with a http url? 

![image](https://user-images.githubusercontent.com/25430059/131695442-b3fe2975-bd5f-4294-9c1e-23fb708ab8d4.png)
Where pdfjsurl is something like http://mydomain/web/assets/pdfjs

I already tried setting the workersrc and the cmapurl 
![image](https://user-images.githubusercontent.com/25430059/131700624-11e45f9a-d0ad-411d-a020-4ed89e3f2e1b.png)
But then I get a 404 on pdf-2.10.544.min.js due to the incorrect url. (pointing to the current application and concat the cdn url isntead of just use the cdn url)

Thanks in advance

**Comment by stephanrauh** [comment]
This setting loads the correct file: 
```typescript
pdfDefaultOptions.workerSrc = () => 'https://unpkg.com/browse/ngx-extended-pdf-viewer@10.0.0-alpha.9/assets/pdf.worker-2.10.550.min.js';
```

Unfortunately, it doesn't work because UNPKG sets the wrong MIME type, but that's another topic.

**Comment by stephanrauh** [comment]
```typescript
pdfDefaultOptions.assetsFolder = 'https://unpkg.com/browse/ngx-extended-pdf-viewer@10.0.0-alpha.9/assets';
```
also seems to work (apart from the error I've mentioned above, which may be caused by my strict origin policy).

**Comment by stephanrauh** [comment]
I can only guess, but I reckon your bug happens in this line:
```typescript
          workerSrc = PDFWorkerUtil.createCDNWrapper(
            new URL(workerSrc, window.location).href   // <<<
          );
```
Please add the parameter `[minifiedJSLibraries]="false"`and set a breakpoint in `api.js` at this code snippet. It's roughly at line 2139, depending on which version you're using. Does `new URL(...).href` result in your broken URLs?

**Reply by JessinhoW** [reply]
Because my cdn is on the same origin as the application that uses the pdfjs lib the code doesn't reach the createcdnwrapper code. So unfortunately it is not working.

**Reply by JessinhoW** [reply]
I fixed it for my situation by modifying ngx-extended-pdf-viewer.js and removing the './' before every line where the pdfDefaultoptions.assetsfolder is used.

**Reply by stephanrauh** [reply]
I see. Maybe that's a special case that's not easy to repair. BTW, I wonder if the bug is related to this bug: https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/813#issuecomment-868495762

**Reply by stephanrauh** [reply]
I just saw you're referring to an actual piece of code (while I suspected the same-origin check to be part of the `new URL().href` snippet):

```js
          if (
            typeof PDFJSDev !== "undefined" &&
            PDFJSDev.test("GENERIC") &&
            !isSameOrigin(window.location.href, workerSrc)
          ) {
            workerSrc = createCDNWrapper(
              new URL(workerSrc, window.location).href
            );
          }
```
